File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.cs
using System;

public class Solution {
    public int[] solution(int denum1, int num1, int denum2, int num2) {

        int parent = num1 * num2;
        int child = denum1 * num2 + denum2 * num1;

        while(true)
        {
            int gcd = parent > child ? GCD(parent, child) : GCD(child, parent);
            if(gcd > 1)
            {
                child /= gcd;
                parent /= gcd;
            }
            else
            {
                break;
            }
        }

        return new int[2] {child, parent};
    }

    public int GCD(int a, int b)
    {
        int r = 0;
        while(b != 0)
        {
            r = a % b;
            a = b;
            b = r;
        }

        return a;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
int GCD(int a, int b);

vector<int> solution(int numer1, int denom1, int numer2, int denom2) {
    
    int child = denom1 * numer2 + denom2 * numer1;
    int parent = denom1 * denom2;
    
    while(true)
    {
        int gcd = child > parent ? GCD(child, parent) : GCD(parent, child);

        if (gcd > 1)
        {
            child /= gcd;
            parent /= gcd;
        }
        else
        {
            break;
        }
    }
    
    return vector<int>{ child, parent };
}

int GCD(int a, int b)
{
    if (b == 0) return a;
    return GCD(b, a % b);
}